src/Neo/SmartContract/KeyBuilder.cs

```diff
@@ -21,15 +21,22 @@ namespace Neo.SmartContract
     /// </summary>
     public class KeyBuilder
     {
-        private readonly MemoryStream stream = new();
+        private readonly MemoryStream stream;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="KeyBuilder"/> class.
         /// </summary>
         /// <param name="id">The id of the contract.</param>
         /// <param name="prefix">The prefix of the key.</param>
-        public KeyBuilder(int id, byte prefix)
+        /// <param name="capacityHint">
+        /// The initial capacity of the stream.
+        /// It is better to set this to the expected size of the key(not including the id and prefix).
+        /// </param>
+        public KeyBuilder(int id, byte prefix, int capacityHint = 0)
```

Comment:
More than capacityInt, byte[] toAdd ? And we save the add and a wrong value here?

Reply:
> More than capacityInt, byte[] toAdd ? And we save the add and a wrong value here?

`id` and `prefix` are always present, so only the part after `prefix` needs to be calculated.

Reply:
We know keys can't exceed 64 bytes (`MaxStorageKeySize`), maybe this can be leveraged to simplify the interface (have static value be used instead of `capacityHint`, it can spend a little more memory, but likely this won't be noticeable.

Reply:
> We know keys can't exceed 64 bytes (`MaxStorageKeySize`), maybe this can be leveraged to simplify the interface (have static value be used instead of `capacityHint`, it can spend a little more memory, but likely this won't be noticeable.

Good idea. It can always allocate 64-bytes.